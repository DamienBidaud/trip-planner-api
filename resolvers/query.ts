import { PrismaClient, PrismaClientOptions, Author } from "@prisma/client";
import { verify } from "jsonwebtoken";
import { getActivity } from "../utils/activity";
type Context = {
  prisma: PrismaClient<PrismaClientOptions, never>;
};

const validateAuth = (_: any, args: { token: string }, context: Context) => {
  const { token } = args;
  try {
    const decoded = verify(token, "DEV_ENV") as {
      authorId: number;
      iat: number;
    };
    return decoded.authorId;
  } catch (err) {
    console.error(err);

    return null;
  }
};

const trip = async (
  _parent: unknown,
  args: { tripId: string },
  context: Context
) => {
  try {
    const allTrips = await context.prisma.trip.findOne({
      where: { id: parseInt(args.tripId, 10) },
    });
    return allTrips;
  } catch (error) {
    console.error(error);
    // throw error;
  }
};

const author = async (_: any, args: { token: string }, context: Context) => {
  const authorId = validateAuth(_, args, context);
  if (authorId) {
    return await context.prisma.author.findOne({
      where: {
        id: authorId,
      },
    });
  }

  return null;
};

const activity = (_: any, args: {activityID: string}, context: Context) =>
  getActivity(parseInt(args.activityID, 10), context);

const locations = (_: any, args: object, context: Context) => {
    return context.prisma.location.findMany();
};

export default {
  validateAuth,
  trip,
  author,
  activity,
  locations
};
