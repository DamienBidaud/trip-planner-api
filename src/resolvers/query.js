const jwt = require("jsonwebtoken");
const { getActivity, getTrip } = require("../utils");

const validateAuth = (_, args, context, info) => {
  const { token } = args;
  try {
    var decoded = jwt.verify(token, process.env.APP_SECRET);
    return decoded.authorId;
  } catch (err) {
    console.error(err);
    return null;
  }
};

const activity = (_, args, context, info) => getActivity(args.activityID, context, info);

const trip = (_, args, context, info) => getTrip(args.tripId, context, info)

const locations = (_, args, context, info) => {
  return context.prisma.query.locations({}, info);
}

const author = (_, args, context, info) => {
  const authorId = validateAuth(_, args, context, info);
  if(!!authorId) {
    return context.prisma.query.author(
      {
        where: {
          id: authorId
        }
      },
      info
    );
  }
  return null;
}

module.exports = {
  validateAuth,
  activity,
  locations,
  author,
  trip
};
