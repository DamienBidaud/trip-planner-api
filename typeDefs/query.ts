import { gql } from "apollo-server";

export default gql`
  type Query {
    validateAuth(token: String!): String
    activity(activityID: ID!): Activity!
    locations: [Location!]!
    author(token: String!): Author
    trip(tripId: ID!): Trip
  }
`;
