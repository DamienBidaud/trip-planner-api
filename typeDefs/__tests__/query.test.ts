import query from "../query";

describe("typeDefs", () => {
  describe("query", () => {
    it("Should match snapshot", () => {
      expect(query).toMatchSnapshot();
    });
  });
});
