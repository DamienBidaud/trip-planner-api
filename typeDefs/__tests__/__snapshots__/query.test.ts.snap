// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`typeDefs query Should match snapshot 1`] = `
Object {
  "definitions": Array [
    Object {
      "description": undefined,
      "directives": Array [],
      "fields": Array [
        Object {
          "arguments": Array [
            Object {
              "defaultValue": undefined,
              "description": undefined,
              "directives": Array [],
              "kind": "InputValueDefinition",
              "name": Object {
                "kind": "Name",
                "value": "token",
              },
              "type": Object {
                "kind": "NonNullType",
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "String",
                  },
                },
              },
            },
          ],
          "description": undefined,
          "directives": Array [],
          "kind": "FieldDefinition",
          "name": Object {
            "kind": "Name",
            "value": "validateAuth",
          },
          "type": Object {
            "kind": "NamedType",
            "name": Object {
              "kind": "Name",
              "value": "String",
            },
          },
        },
        Object {
          "arguments": Array [
            Object {
              "defaultValue": undefined,
              "description": undefined,
              "directives": Array [],
              "kind": "InputValueDefinition",
              "name": Object {
                "kind": "Name",
                "value": "activityID",
              },
              "type": Object {
                "kind": "NonNullType",
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "ID",
                  },
                },
              },
            },
          ],
          "description": undefined,
          "directives": Array [],
          "kind": "FieldDefinition",
          "name": Object {
            "kind": "Name",
            "value": "activity",
          },
          "type": Object {
            "kind": "NonNullType",
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "Activity",
              },
            },
          },
        },
        Object {
          "arguments": Array [],
          "description": undefined,
          "directives": Array [],
          "kind": "FieldDefinition",
          "name": Object {
            "kind": "Name",
            "value": "locations",
          },
          "type": Object {
            "kind": "NonNullType",
            "type": Object {
              "kind": "ListType",
              "type": Object {
                "kind": "NonNullType",
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "Location",
                  },
                },
              },
            },
          },
        },
        Object {
          "arguments": Array [
            Object {
              "defaultValue": undefined,
              "description": undefined,
              "directives": Array [],
              "kind": "InputValueDefinition",
              "name": Object {
                "kind": "Name",
                "value": "token",
              },
              "type": Object {
                "kind": "NonNullType",
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "String",
                  },
                },
              },
            },
          ],
          "description": undefined,
          "directives": Array [],
          "kind": "FieldDefinition",
          "name": Object {
            "kind": "Name",
            "value": "author",
          },
          "type": Object {
            "kind": "NamedType",
            "name": Object {
              "kind": "Name",
              "value": "Author",
            },
          },
        },
        Object {
          "arguments": Array [
            Object {
              "defaultValue": undefined,
              "description": undefined,
              "directives": Array [],
              "kind": "InputValueDefinition",
              "name": Object {
                "kind": "Name",
                "value": "tripId",
              },
              "type": Object {
                "kind": "NonNullType",
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "ID",
                  },
                },
              },
            },
          ],
          "description": undefined,
          "directives": Array [],
          "kind": "FieldDefinition",
          "name": Object {
            "kind": "Name",
            "value": "trip",
          },
          "type": Object {
            "kind": "NamedType",
            "name": Object {
              "kind": "Name",
              "value": "Trip",
            },
          },
        },
      ],
      "interfaces": Array [],
      "kind": "ObjectTypeDefinition",
      "name": Object {
        "kind": "Name",
        "value": "Query",
      },
    },
  ],
  "kind": "Document",
  "loc": Object {
    "end": 193,
    "start": 0,
  },
}
`;
